fuse_examples/classification/multimodality/mg_dataset.py

```diff
@@ -0,0 +1,432 @@
+        self.continuous_tabular_features_lst = continuous_tabular_features_lst
+        self.categorical_tabular_features_lst = categorical_tabular_features_lst
+        self.label_lst = label_lst
+        self.imaging_features_lst = imaging_features_lst
+        self.non_imaging_features_lst = non_imaging_features_lst
+        self.use_imaging = use_imaging
+        self.use_non_imaging = use_non_imaging
+
+    def __call__(self, batch_dict: Dict) -> Dict:
+        if not self.use_imaging and not self.use_non_imaging:
+            raise ValueError('No features are in use')
+        mask_list = self.use_imaging * self.imaging_features_lst + self.use_non_imaging * self.non_imaging_features_lst
+        mask_continuous = torch.zeros(len( self.continuous_tabular_features_lst))
+        for i in range(len(mask_list)):
+            try:
+                mask_continuous[self.continuous_tabular_features_lst.index(mask_list[i])] = 1
+            except:
+                pass
+        mask_categorical = torch.zeros(len( self.categorical_tabular_features_lst))
+        for i in range(len(mask_list)):
+            try:
+                mask_categorical[self.categorical_tabular_features_lst.index(mask_list[i])] = 1
+            except:
+                pass
+        categorical = [FuseUtilsHierarchicalDict.get(batch_dict, 'data.' + feature_name) for feature_name in self.categorical_tabular_features_lst]
+        for i in range(len(categorical)):
+            if categorical[i].dim() == 0:
+                categorical[i] = torch.unsqueeze(categorical[i], 0)
+        categorical_tensor = torch.cat(tuple(categorical), 0)
+        categorical_tensor = categorical_tensor.float()
+        categorical_tensor = torch.mul(categorical_tensor, mask_categorical)
+        FuseUtilsHierarchicalDict.set(batch_dict, 'data.categorical', categorical_tensor.float())
+        continuous = [FuseUtilsHierarchicalDict.get(batch_dict, 'data.' + feature_name) for feature_name in self.continuous_tabular_features_lst]
+        for i in range(len(continuous)):
+            if continuous[i].dim() == 0:
+                continuous[i] = torch.unsqueeze(continuous[i], 0)
+        continuous_tensor = torch.cat(tuple(continuous), 0)
+        continuous_tensor = continuous_tensor.float()
+        continuous_tensor = torch.mul(continuous_tensor, mask_continuous)
+        FuseUtilsHierarchicalDict.set(batch_dict, 'data.continuous', continuous_tensor.float())
+        label = FuseUtilsHierarchicalDict.get(batch_dict, 'data.' + self.label_lst[0])
+        FuseUtilsHierarchicalDict.set(batch_dict, 'data.label', label.long())
+        feature_lst = self.continuous_tabular_features_lst + self.categorical_tabular_features_lst + self.label_lst
+        for feature in feature_lst:
+            FuseUtilsHierarchicalDict.pop(batch_dict, 'data.' + feature)
+        return batch_dict
+
+
+# feature selection univarient analysis
+
+def get_selected_features_clinical(output_path):
```

Comment:
Are you using the code from this line forward? seems like pre-dataset creation code.